File: Bit_Manipulation/Check_Pow_2.py
# Check whether a given number n is a power of 2 or 0


def check_pow_2(num):
    if num == 0:
        return 0

    if num & (num - 1) == 0:
        return 1

    return -1


switch = {
    0: "Number is 0",
    1: "Number is a power of 2",
    -1: "Number is neither a power of 2 nor 0"
}
case = check_pow_2(16)

print(switch[case])

File: Bit_Manipulation/Check_Pow_2.cpp
/*
Check whether a given number n is a power of 2
*/

#include <iostream>

using namespace std;

bool check_pow_2(int num) {
    // num is a power of 2 only if its binary representation has one bit as 1.
    if ((num & (num-1)) == 0)
    	return true;
    return false;
}

int main() {
    int num = 2;

    check_pow_2(num) ? cout << num << " is a Power of 2" : cout << num << " is not a power of 2";
    cout << endl;

    num = 5;
    check_pow_2(num) ? cout << num << " is a Power of 2" : cout << num << " is not a power of 2";
    cout << endl;

    return 0;
}
